encode.avs

```diff
@@ -169,22 +168,35 @@ hARC	= height
 
 #	Actually go HD if we need
 if (hd) {
-	height = hdHeight
-}
-
-width	= (ARCorrection \
-		? height * wAspect / hAspect \
-		: height * last.width / last.height) \
-		.ForceModulo(mod, true)
-hStretch= height / last.height
-
-# YouTube has a 4K width limit of 3840
-if (hd && (width > 3840)) {
-	width = 3840
-	height = (ARCorrection \
-		? width * hAspect / wAspect \
-		: width * last.height / last.width) \
-		.ForceModulo(mod, true)
+	if (width >= height) {
+		height = 2160
+		width	= (ARCorrection \
+			? height * wAspect / hAspect \
+			: height * last.width / last.height) \
+			.ForceModulo(mod, true)
+		# YouTube has a 4K width limit of 3840 when width > height
+		if (width > 3840) {
+			width = 3840
+			height = (ARCorrection \
+				? width * hAspect / wAspect \
+				: width * last.height / last.width) \
+				.ForceModulo(mod, true)
+		}
+	} else {
+		width = 2160
+		height	= (ARCorrection \
+			? width * hAspect / wAspect \
+			: width * last.height / last.width) \
+			.ForceModulo(mod, true)
+		# YouTube has a 4K height limit of 3840 when height > width
+		if (height > 3840) {
+			height = 3840
+			width = (ARCorrection \
+				? height * wAspect / hAspect \
+				: height * last.width / last.height) \
+				.ForceModulo(mod, true)
+		}
+	}
 }

```

Comment:
Can you check if this code would work as a replacement?

```
if (hd) {
	vertical        = width < height
	
	normalAspect	= ARCorrection ? (float(wAspect) / hAspect) : (last.width / last.height)
	reverseAspect	= ARCorrection ? (float(hAspect) / wAspect) : (last.height / last.width)
	
	smallerSide		= vertical ? width  : height
	biggerSide		= vertical ? height : width
	
	smallerSideCap  = 2160
	biggerSideCap   = 3840
	
	smallerSideHD	= smallerSideCap
	biggerSideHD	= (smallerSideHD * normalAspect).ForceModulo(mod, true)
	
	if (biggerSideHD > biggerSideCap) {
		biggerSideHD	= biggerSideCap
		smallerSideHD	= biggerSideHD * reverseAspect).ForceModulo(mod, true)
	}
	
	height	= vertival ? smallerSideHD : biggerSideHD
	width	= vertical ? biggerSideHD  : smallerSideHD
}
```

Reply:
Fixed up some bugs and added more checks on some use cases.